baseapp/abci.go

```diff
@@ -74,6 +74,16 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 		}
 	}
 
+	defer func() {
```

Comment:
`defer` is usually for resource cleanup, I think it's clearer to simply put the code after the commit call.

Reply:
Although the tutorial on defer states that, there's no idiomatic usage of it.

> defer is often used where e.g. ensure and finally would be used in other languages.

Finally is exactly what we're trying to do here. I think if we do it in `Commit`, we'll have to explicitly check if height == 0.

Reply:
I mean just after the initChainer call, I think we don't want to run this if there's any errors or panics?

Reply:
ahh yes, the reason why I added in the defer is if `initChainer == nil`, you'll need to add it before the return. That's why I added in the defer instead of having it twice.